### PR TITLE
feature: enhance vip and resolve conflict when merge release-0.4.0 to main

### PR DESCRIPTION
**Pull request by village-way**
#### What type of PR is this?

/kind bug
/kind cleanup
#### Which issue(s) does this PR fix?

Fixes #
#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: deploy/virtual-cluster-components-manifest-cm.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@ data:
     [
       {"name": "kube-proxy", "path": "/kosmos/manifest/kube-proxy/*.yaml"},
       {"name": "calico", "path": "/kosmos/manifest/calico/*.yaml"},
+      {"name": "keepalived", "path": "/kosmos/manifest/keepalived/*.yaml"},
     ]
```
Comment (OrangeBao) [inline deploy/virtual-cluster-components-manifest-cm.yaml:8]: It is necessary to emphasize in the deployment document or create a separate chapter to introduce how to deploy VIP.

File: pkg/apis/kosmos/v1alpha1/virtualcluster_types.go [hunk 2]
```diff
@@ -56,6 +56,10 @@ type VirtualClusterSpec struct {
 	// +optional
 	ExternalIps []string `json:"externalIps,omitempty"`
 
+	// ExternalVips is the external vips of the virtual kubernetes's control plane
+	// +optional
+	ExternalVips []string `json:"externalVips,omitempty"`
+
 	// PromotePolicies definites the policies for promote to the kubernetes's control plane
```
Comment (OrangeBao) [inline pkg/apis/kosmos/v1alpha1/virtualcluster_types.go:62]: Group and classify these parameters in advance

File: deploy/virtual-cluster-components-manifest-cm.yaml (hunk 1 above)
Comment (village-way) [reply deploy/virtual-cluster-components-manifest-cm.yaml:8]: add simple quick start guide in conf

File: pkg/apis/kosmos/v1alpha1/virtualcluster_types.go (hunk 2 above)
Comment (village-way) [reply pkg/apis/kosmos/v1alpha1/virtualcluster_types.go:62]: updated with follow code
```
	// KubeInKubeConfig is the external config of virtual cluster
	// +optional
	KubeInKubeConfig KubeInKubeConfig `json:"kubeInKubeConfig,omitempty"`
```

```

type KubeInKubeConfig struct {
	// todo Group according to the parameters of apiserver, etcd, coredns, etc.
	// ForceDestroy indicates whether to force destroy the cluster.
	// +optional
	ForceDestroy bool `yaml:"forceDestroy" json:"forceDestroy,omitempty"`
	// +optional
	AnpMode string `yaml:"anpMode" json:"anpMode,omitempty"`
	// +optional
	AdmissionPlugins bool `yaml:"admissionPlugins" json:"admissionPlugins,omitempty"`
	// +optional
	ApiServerReplicas int `yaml:"apiServerReplicas" json:"apiServerReplicas,omitempty"`
	// +optional
	ClusterCIDR string `yaml:"clusterCIDR" json:"clusterCIDR,omitempty"`
	// +optional
	ETCDStorageClass string `yaml:"etcdStorageClass" json:"etcdStorageClass,omitempty"`
	// +optional
	ETCDUnitSize string `yaml:"etcdUnitSize" json:"etcdUnitSize,omitempty"`

	//// Etcd contains the configuration for the etcd statefulset.
	//Etcd EtcdCluster `yaml:"etcd" json:"etcd,omitempty"`

	//// DNS  contains the configuration for the dns server in kubernetes.
	//DNS DNS `yaml:"dns" json:"dns,omitempty"`
	//
	//// Kubernetes contains the configuration for the kubernetes.
	//Kubernetes Kubernetes `yaml:"kubernetes" json:"kubernetes,omitempty"`
	//
	//// Network  contains the configuration for the network in kubernetes cluster.
	//Network NetworkConfig `yaml:"network" json:"network,omitempty"`
	//
	//// Storage contains the configuration for the storage in kubernetes cluster.
	//Storage StorageConfig `yaml:"storage" json:"storage,omitempty"`
	//
	//// Registry contains the configuration for the registry in kubernetes cluster.
	//Registry RegistryConfig `yaml:"registry" json:"registry,omitempty"`

	//TenantEntrypoint TenantEntrypoint `yaml:"tenantEntrypoint" json:"tenantEntrypoint,omitempty"`
	// +optional
	TenantEntrypoint TenantEntrypoint `yaml:"tenantEntrypoint" json:"tenantEntrypoint,omitempty"`
}

// TenantEntrypoint contains the configuration for the tenant entrypoint.
type TenantEntrypoint struct {
	// ExternalIP is the external ip of the tenant entrypoint.
	// +optional
	ExternalIps []string `json:"externalIps,omitempty"`
	// ExternalVips is the external vips of the tenant entrypoint.
	// +optional
	ExternalVips []string `json:"externalVips,omitempty"`
}

```

**Pull request merged**